pkg/highlight/highlighter.go

```diff
@@ -283,11 +342,22 @@ func (h *Highlighter) highlight(highlights LineMatch, start int, lineNum int, li
 		return highlights
 	}
 
-	h.lastStart = 0
-	h.lastEnd = 0
+	h.lastStart = -1
+	h.lastEnd = -1
+	h.storage = h.storage[:0]
+
+	h.storeRegions(start, lineNum, line, curRegion, h.Def.rules.regions, false)
 
 	fullHighlights := make([]Group, lineLen)
-	h.highlightRegions(fullHighlights, start, lineNum, line, curRegion, h.Def.rules.regions, false)
+
+	for _, e := range h.storage {
+		if e.start <= e.end && e.end <= len(fullHighlights) {
```

Comment:
Seems like it was added out of paranoia to prevent crossing the boundary, but in even then it would hide issues in the previous highlighter code, which should be found.
I removed it, opened all my test files and there was no crash. So seems to be OK for now.

Thanks for spotting. :+1: